internal/controller/mapserver/deployment.go

```diff
@@ -251,8 +249,16 @@ func GetResourcesForDeployment[O pdoknlv3.WMSWFS](obj O) v1.ResourceRequirements
 		resources.Limits[v1.ResourceCPU] = *objResources.Limits.Cpu()
 	}
 
-	if objResources.Requests.Memory() != nil && !objResources.Requests.Memory().IsZero() {
-		resources.Requests[v1.ResourceMemory] = *objResources.Requests.Memory()
+	memoryRequest := objResources.Requests.Memory()
+	if memoryRequest != nil && !memoryRequest.IsZero() {
+		resources.Requests[v1.ResourceMemory] = *memoryRequest
+	}
+
+	memoryLimit := objResources.Limits.Memory()
+	if memoryLimit != nil && !memoryLimit.IsZero() {
```

Comment:
Ik zou toevoegen dat als de request bestaat en  hoger is dan de limit dat dan de request wordt gebruikt